### PR TITLE
[SPARK-37413][PYTHON] Inline type hints for python/pyspark/ml/tree.py

### PR DESCRIPTION
**Pull request by dchvn**
### What changes were proposed in this pull request?
Inline type hints for python/pyspark/ml/tree.py

### Why are the changes needed?
We can take advantage of static type checking within the functions by inlining the type hints.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by dchvn** [comment]
cc @zero323

File: python/pyspark/ml/tree.py [hunk 1]
```diff
@@ -300,52 +307,52 @@ class _RandomForestParams(_TreeEnsembleParams):
...
-    numTrees = Param(
+    numTrees: Param[int] = Param(
         Params._dummy(),
         "numTrees",
         "Number of trees to train (>= 1).",
         typeConverter=TypeConverters.toInt,
     )
 
-    bootstrap = Param(
+    bootstrap: Param[int] = Param(
```
Comment (zero323) [inline python/pyspark/ml/tree.py:317]: Should be `Param[bool]`

**Review by zero323 — COMMENTED** [review]
Let's make sure this is still compatible with master (performed local test and it looked OK, but we already had some hiccups with things getting out of sync).

File: python/pyspark/ml/tree.py (hunk 1 above)
Comment (dchvn) [reply python/pyspark/ml/tree.py]: Updated, thanks!

**Review by zero323 — APPROVED** [review]

**Comment by zero323** [comment]
Merged into master.

**Comment by dchvn** [comment]
Thanks 😄

**Pull request closed without merging**